File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
while True:
    a, b, c = map(int, input().split())
    lst = [a, b, c]

    if a == b == c == 0:
        break

    if max(a, b, c) >= sum(lst)-max(a, b, c):
        print('Invalid')

    elif len(set(lst)) == 1:
        print('Equilateral')

    elif len(set(lst)) == 2:
        print('Isosceles')

    elif len(set(lst)) == 3:
        print('Scalene')

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(NULL);
  int x, y, z;

  while (cin >> x >> y >> z) {
    if (x == 0 && y == 0 && z == 0) {
      break;
    }

    if (x+y+z - (2*max({x, y, z})) <= 0) {
      cout << "Invalid" << '\n';
      continue;
    }

    if (x == y && y == z) {
      cout << "Equilateral" << '\n';
    }
    else if (x == y || y == z || x == z) {
      cout << "Isosceles" << '\n';
    }
    else {
      cout << "Scalene" << '\n';
    }
  }

  return 0;
}